client/app/components/items-list/classes/StateStorage.js

```diff
@@ -0,0 +1,60 @@
+import { defaults } from 'lodash';
+import { clientConfig } from '@/services/auth';
+import { $location } from '@/services/ng';
+import { parse as parseOrderBy, compile as compileOrderBy } from './Sorter';
+
+export class StateStorage {
+  constructor(state = {}) {
+    this._state = { ...state };
+  }
+
+  getState() {
+    return defaults(this._state, {
+      page: 1,
+      itemsPerPage: clientConfig.pageSize,
+      orderByField: 'created_at',
+      orderByReverse: false,
+      searchTerm: '',
+      tags: [],
+    });
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  setState() {
+  }
+}
+
+export class UrlStateStorage extends StateStorage {
+  getState() {
+    const defaultState = super.getState();
+    const params = $location.search();
+
+    const searchTerm = params.q || '';
+
+    // in search mode order by should be explicitly specified in url, otherwise use default
+    const defaultOrderBy = searchTerm !== '' ? '' : compileOrderBy(defaultState.orderByField, defaultState.orderByReverse);
+
+    const {
+      field: orderByField,
+      reverse: orderByReverse,
+    } = parseOrderBy(params.order || defaultOrderBy);
+
+    return {
+      page: parseInt(params.page, 10) || defaultState.page,
```

Comment:
@kravets-levko wdyt of generally using `Number(str)` instead?

Reply:
Can't decide right now :thinking: I used to use `parseInt`/`parseFloat` (no particular reason why I prefer it).

Reply:
For our usage it has no real difference in output (except `Number('')` -> `0`, while `parseInt('', 10)` -> `NaN`).
`parseInt` comes with extra features that are irrelevant.

Reply:
Ah, I realized why `parseInt` here :slightly_smiling_face: Because `Number()` constructor is in fact `parseFloat`, so `Number('3.14') === 3.14`, but `parseInt('3.14') === 3`

Reply:
And probably that's also the answer why I prefer `parseInt`/`parseFloat` - it's easy to forget that `Number()` will parse string as floating point value, and when using functions you explicitly see what you're doing.